SalesforceSDK/Analytics/Store/EventStoreManager.cs

```diff
@@ -0,0 +1,224 @@
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using Newtonsoft.Json.Linq;
+using Salesforce.SDK.Analytics.Model;
+using Salesforce.SDK.Core;
+using Salesforce.SDK.Logging;
+using PCLStorage;
+
+namespace Salesforce.SDK.Analytics.Store
+{
+    public class EventStoreManager : IEventStoreManager
+    {
+        private string _filenameSuffix;
+        private readonly IFolder _rootDir;
+        private string _encryptionKey;
+        private bool _isLoggingEnabled = true;
+        private int _maxEvents = 1000;
+        private static ILoggingService LoggingService => SDKServiceLocator.Get<ILoggingService>();
+        public string FilenameSuffix
+        {
+            get { return _filenameSuffix; }
+            set { _filenameSuffix = value; }
+        } 
+
+        public string EncryptionKey
+        {
+            get { return _encryptionKey; }
+            set { _encryptionKey = value; }
+        }
+
+        public EventStoreManager(string fileNameSuffix, string encryptionKey)
+        {
+            FilenameSuffix = fileNameSuffix;
+            EncryptionKey = encryptionKey;
+            _rootDir = FileSystem.Current.LocalStorage;
+        }
+
+        public async Task StoreEventAsync(InstrumentationEvent instrumentationEvent)
+        {
+            if (instrumentationEvent == null || string.IsNullOrEmpty(instrumentationEvent.ToJson().ToString()))
+            {
+                LoggingService.Log("Invalid Event", LoggingLevel.Error);
+                return;
+            }
+            //TODO: Add check for shouldstoreevent
+            var fileName = instrumentationEvent.EventId + _filenameSuffix;
+            //Open file
+            IFile file = await _rootDir.CreateFileAsync(fileName, CreationCollisionOption.OpenIfExists);
+            //Wrtie to file after encrypting contents
+            //TODO: Enrypt the contents
+            await file.WriteAllTextAsync(instrumentationEvent.ToJson().ToString());
+        }
+
+        public async Task StoreEventsAsync(List<InstrumentationEvent> instrumentationEvents)
+        {
+            if (instrumentationEvents == null || instrumentationEvents.Count == 0)
+            {
+                LoggingService.Log("No events to store", LoggingLevel.Error);
+                return;
+            }
+            if (!await ShouldStoreEvent())
+            {
+                return;
+            }
+            foreach (var instrumentationEvent in instrumentationEvents)
+            {
+                await StoreEventAsync(instrumentationEvent);
```

Comment:
do these events need to be stored serially?  Instead of awaiting in a for loop you could let them all run the StoreEventAsync code in parallel.  Just replace your foreach loop with something like `Task.WhenAll(instrumentationEvents.Select(StoreEventAsync);`